web/modules/custom/custom_page/src/Controller/CustomPageController.php

```diff
@@ -59,7 +56,28 @@ public function createCustomPageAccess(RdfInterface $rdf_entity) {
       return AccessResult::allowed();
     }
 
+    // Check if the user has the OG permission to create a custom page.
+    if (OgAccess::userAccessGroupContentEntityCrud('create', $rdf_entity, $this->createNewCustomPage($rdf_entity))->isAllowed()) {
```

Comment:
The first check is intended for moderators, here we need to check the bundle, because the global permission is not granular per bundle.

The second check does not need an additional check on the bundle, we are passing the `$rdf_entity` to `OgAccess::userAccessGroupContentEntityCrud()` which has per-bundle permissions. It will get the bundle from the entity and will check if the user has this permission within the bundle. So if the bundle is not a collection then it will not return `allowed` (unless the user also happens to have access in that other group type).